webapp/src/components/token-usage/TokenUsageGraph.tsx

```diff
@@ -0,0 +1,159 @@
+import {
+    Body1,
+    Button,
+    Divider,
+    makeStyles,
+    mergeClasses,
+    Popover,
+    PopoverSurface,
+    PopoverTrigger,
+    shorthands,
+    Text,
+    tokens,
+} from '@fluentui/react-components';
+import { Brands } from '@fluentui/tokens';
+import {
+    TokenUsage,
+    TokenUsageFunctionNameMap,
+    TokenUsageView,
+    TokenUsageViewDetails,
+} from '../../libs/models/TokenUsage';
+import { useAppSelector } from '../../redux/app/hooks';
+import { RootState } from '../../redux/app/store';
+import { semanticKernelBrandRamp } from '../../styles';
+import { TypingIndicator } from '../chat/typing-indicator/TypingIndicator';
+import { Info16 } from '../shared/BundledIcons';
+import { TokenUsageBar } from './TokenUsageBar';
+import { TokenUsageLegendItem } from './TokenUsageLegendItem';
+
+const useClasses = makeStyles({
+    horizontal: {
+        display: 'flex',
+        ...shorthands.gap(tokens.spacingVerticalSNudge),
+        alignItems: 'center',
+    },
+    content: {
+        display: 'flex',
+        flexDirection: 'column',
+        ...shorthands.gap(tokens.spacingHorizontalS),
+        paddingBottom: tokens.spacingHorizontalM,
+    },
+    popover: {
+        width: '300px',
+    },
+    header: {
+        marginBlockEnd: tokens.spacingHorizontalM,
+    },
+    legend: {
+        'flex-flow': 'wrap',
+    },
+});
+
+interface ITokenUsageGraph {
+    tokenUsage: TokenUsage;
+    promptView?: boolean;
+}
+
+export const TokenUsageGraph: React.FC<ITokenUsageGraph> = ({ promptView, tokenUsage }) => {
+    const classes = useClasses();
+    const { conversations, selectedId } = useAppSelector((state: RootState) => state.conversations);
+    const loadingResponse = conversations[selectedId].botResponseStatus;
+
+    const tokenUsageView: TokenUsageView = {};
+    let dependencyUsage = 0;
+    let responseGenerationUsage = 0;
+    let color = 160 as Brands;
+    Object.entries(tokenUsage).forEach(([key, value]) => {
+        const viewDetails: TokenUsageViewDetails = {
+            usageCount: value ?? 0,
+            legendLabel: TokenUsageFunctionNameMap[key],
+            dependency: false,
+            color: semanticKernelBrandRamp[color],
+        };
+
+        // Iterate through the color ramp
+        color -= 20;
```

Comment:
nit: prevent it to go out of bound
```suggestion
        color = (color - 20 < 0) ? 160 : color - 20;
```